tools/hula/proxy/src/proxy.rs

```diff
@@ -0,0 +1,222 @@
+    let since_start = proxy_start.elapsed();
+    let mut robot_state = read_lola_message(lola).wrap_err("failed to read lola message")?;
+    robot_state.received_at = since_start.as_secs_f32();
+    *battery = Some(robot_state.battery);
+
+    if connections.is_empty() {
+        return Ok(());
+    }
+    let state_storage_buffer = unsafe {
+        from_raw_parts(
+            &robot_state as *const RobotState as *const u8,
+            size_of::<RobotState>(),
+        )
+    };
+    connections.retain(|_, connection| {
+        if let Err(error) = connection.socket.write_all(state_storage_buffer) {
+            error!("Failed to write StateStorage to connection: {error}");
+            return false;
+        }
+        if let Err(error) = connection.socket.flush() {
+            error!("Failed to flush connection: {error}");
+            return false;
+        }
+        true
+    });
+    Ok(())
+}
+
+fn read_lola_message(lola: &mut UnixStream) -> Result<RobotState> {
+    let mut lola_data = [0; 896];
+    lola.read_exact(&mut lola_data)
+        .wrap_err("failed to read from LoLA socket")?;
+    from_slice(&lola_data).wrap_err("failed to parse MessagePack from LoLA StateMessage")
+}
+
+fn register_connection(
+    hula: &mut HulaListener,
+    connections: &mut HashMap<RawFd, Connection>,
+    poll_fd: RawFd,
+) -> Result<()> {
+    let (connection_stream, _) = hula.accept().wrap_err("failed to accept connection")?;
+    let connection_fd = connection_stream.as_raw_fd();
+    info!("Accepted connection with file descriptor {connection_fd}");
+    if connections
+        .insert(
+            connection_fd,
+            Connection {
+                socket: connection_stream,
+                last_time_received: None,
+            },
+        )
+        .is_some()
+    {
+        panic!("connection is already registered");
+    }
+    add_to_epoll(poll_fd, connection_fd)
+        .wrap_err("failed to register connection file descriptor")?;
+
+    Ok(())
+}
+
+fn handle_connection_event(
+    connections: &mut HashMap<RawFd, Connection>,
+    notified_fd: RawFd,
+    writer: &mut BufWriter<UnixStream>,
+    battery: &Option<Battery>,
+) -> Result<()> {
+    match connections.get_mut(&notified_fd) {
+        Some(connection) => {
+            let mut read_buffer = [0; size_of::<HulaControlFrame>()];
+            if let Err(error) = connection.socket.read_exact(&mut read_buffer) {
+                error!("Failed to read from connection: {}", error);
+                info!("Removing connection with file descriptor {}", notified_fd);
+                // remove will drop, drop will close, close will EPOLL_CTL_DEL
+                connections
+                    .remove(&notified_fd)
+                    .expect("connection file descriptor has to be registered");
+                return Ok(());
+            };
+            connection.last_time_received = Some(Instant::now());
+            // reinterpret the read buffer as a ControlFrame
```

Comment:
Unnecessary comment? This is what the next line states?

Reply:
removed